hack/preview.sh

```diff
@@ -125,30 +119,52 @@ git checkout $MY_GIT_BRANCH
 #set the local cluster to point to the current git repo and branch and update the path to development
 $ROOT/hack/util-update-app-of-apps.sh $MY_GIT_REPO_URL development $PREVIEW_BRANCH
 
+while [ "$(oc get applications.argoproj.io all-components-staging -n openshift-gitops -o jsonpath='{.status.health.status} {.status.sync.status}')" != "Healthy Synced" ]; do
+  sleep 5
+done
+
+APPS=$(kubectl get apps -n openshift-gitops -o name)
+
+if echo $APPS | grep -q spi; then
+  if [ "`oc get applications.argoproj.io spi -n openshift-gitops -o jsonpath='{.status.health.status} {.status.sync.status}'`" != "Healthy Synced" ]; then
+    echo Initializing SPI
+    curl https://raw.githubusercontent.com/redhat-appstudio/e2e-tests/main/scripts/spi-e2e-setup.sh | bash -s
```

Comment:
@Michkov I just realized that in case we'd need to update this script in e2e-tests repo, we should be able to test the change somehow.

Openshift CI exports an env var called `$PULL_PULL_SHA` which contains a SHA of a latest PR commit, which we could pass to `preview.sh` script (and probably rename the env var to something more appropriate, like `$E2E_TESTS_COMMIT_SHA`). Thanks to that we could test the change introduced in the e2e-tests PR.

Would it make sense to update this line accordingly?
```bash
curl https://raw.githubusercontent.com/redhat-appstudio/e2e-tests/${E2E_TESTS_COMMIT_SHA:-main}/scripts/spi-e2e-setup.sh | bash -s
```

Reply:
updated